src/model/channel/attachment_type.rs

```diff
@@ -0,0 +1,81 @@
+use std::borrow::Cow;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+/// Enum that allows a user to pass a [`Path`] or a [`File`] type to [`send_files`]
+///
+/// [`send_files`]: crate::model::id::ChannelId::send_files
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum AttachmentType<'a> {
```

Comment:
Changing the location of a type is a major breaking change, which is forbidden on the `current` branch. You will have to retarget your PR to the `next` branch.